src/librustc_mir/build/expr/as_rvalue.rs

```diff
@@ -125,35 +125,36 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                 //     to the same MIR as `let x = ();`.
 
                 // first process the set of fields
-                let fields: Vec<_> =
-                    fields.into_iter()
-                          .map(|f| unpack!(block = this.as_operand(block, f)))
-                          .collect();
+                let fields: Vec<_> = fields.into_iter()
+                                           .map(|f| unpack!(block = this.as_operand(block, f)))
+                                           .collect();
 
                 block.and(Rvalue::Aggregate(AggregateKind::Vec, fields))
             }
             ExprKind::Tuple { fields } => { // see (*) above
                 // first process the set of fields
-                let fields: Vec<_> =
-                    fields.into_iter()
-                          .map(|f| unpack!(block = this.as_operand(block, f)))
-                          .collect();
+                let fields: Vec<_> = fields.into_iter()
+                                           .map(|f| unpack!(block = this.as_operand(block, f)))
+                                           .collect();
 
                 block.and(Rvalue::Aggregate(AggregateKind::Tuple, fields))
             }
             ExprKind::Closure { closure_id, substs, upvars } => { // see (*) above
-                let upvars =
-                    upvars.into_iter()
-                          .map(|upvar| unpack!(block = this.as_operand(block, upvar)))
-                          .collect();
+                let upvars = upvars.into_iter()
+                                   .map(|upvar| unpack!(block = this.as_operand(block, upvar)))
+                                   .collect();
                 block.and(Rvalue::Aggregate(AggregateKind::Closure(closure_id, substs), upvars))
             }
             ExprKind::Adt { adt_def, variant_index, substs, fields, base } => { // see (*) above
                 // first process the set of fields
-                let fields_map: FnvHashMap<_, _> =
-                    fields.into_iter()
-                          .map(|f| (f.name, unpack!(block = this.as_operand(block, f.expr))))
-                          .collect();
+                let fields_map: FnvHashMap<_, _> = fields.into_iter()
```

Comment:
Yeah, it's something I've seen. I agree it is sub-optimal. I think the simple idea of comparing with or without line breaks after the `=` in assignments is probably a good idea - the majority of times I see this is in a `let` statement, so it would probably get us 90% of the way there. I worry a bit about combinatorial explosion - we'd want to do the check in as few places as possible.